### ISSUE TITLE
Switch expression not handled when used as return value

### ISSUE THREAD

**Issue by dododge** [uber/NullAway#478]
Using NullAway 0.9.1 with error-prone 2.6.0. When a method tries to return a value directly from a switch expression, I get an exception:

```
[ERROR]      BugPattern: NullAway
[ERROR]      Stack Trace:
[ERROR]      java.lang.RuntimeException: whoops, better handle SWITCH_EXPRESSION switch (val) {
[ERROR]               case 0 -> "zero";
[ERROR]               case 1 -> "one";
[ERROR]               default -> "other";
[ERROR]           }
[ERROR]   	at com.uber.nullaway.NullAway.mayBeNullExpr(NullAway.java:1946)
[ERROR]   	at com.uber.nullaway.NullAway.checkReturnExpression(NullAway.java:629)
[ERROR]   	at com.uber.nullaway.NullAway.matchReturn(NullAway.java:292)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:450)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitReturn(ErrorProneScanner.java:817)
[ERROR]   	at com.google.errorprone.scanner.ErrorProneScanner.visitReturn(ErrorProneScanner.java:151)
[ERROR]   	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCReturn.accept(JCTree.java:1665)
[ERROR]   	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:86)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:74)
[ERROR]   	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:48)
...
```

A simple class that triggers the problem:

```java
package example;

public class Example
{
    public static String example(int val) {
        return switch (val) {
            case 0 -> "zero";
            case 1 -> "one";
            default -> "other";
        };
    }
}
```

**Comment by msridhar** [comment]
Duplicate of #289

**Comment by msridhar** [comment]
@dododge we are still blocked on Checker Framework support for switch expressions.  Unless they get an external contribution, I don't expect them to add support until after the release of JDK 17.  We'll update #289 as we get more info
